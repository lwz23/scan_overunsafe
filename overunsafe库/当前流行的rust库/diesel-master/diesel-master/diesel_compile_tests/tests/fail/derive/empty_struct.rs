#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
    }
}

#[derive(AsChangeset)]
#[diesel(table_name = users)]
struct User1;

#[derive(Identifiable)]
#[diesel(table_name = users)]
struct User2;

#[derive(Insertable)]
#[diesel(table_name = users)]
struct User3;

#[derive(Queryable)]
struct User4;

#[derive(QueryableByName)]
struct User5;

#[derive(Selectable)]
struct User6;

#[derive(Associations)]
#[diesel(table_name = users)]
struct User7;

fn main() {}
