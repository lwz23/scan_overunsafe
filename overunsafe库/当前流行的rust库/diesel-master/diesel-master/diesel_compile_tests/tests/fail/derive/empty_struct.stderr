error: This derive can only be used on non-unit structs
  --> tests/fail/derive/empty_struct.rs:10:10
   |
10 | #[derive(AsChangeset)]
   |          ^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `AsChangeset` (in Nightly builds, run with -Z macro-backtrace for more info)

error: This derive can only be used on non-unit structs
  --> tests/fail/derive/empty_struct.rs:14:10
   |
14 | #[derive(Identifiable)]
   |          ^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `Identifiable` (in Nightly builds, run with -Z macro-backtrace for more info)

error: This derive can only be used on non-unit structs
  --> tests/fail/derive/empty_struct.rs:18:10
   |
18 | #[derive(Insertable)]
   |          ^^^^^^^^^^
   |
   = note: this error originates in the derive macro `Insertable` (in Nightly builds, run with -Z macro-backtrace for more info)

error: This derive can only be used on non-unit structs
  --> tests/fail/derive/empty_struct.rs:22:10
   |
22 | #[derive(Queryable)]
   |          ^^^^^^^^^
   |
   = note: this error originates in the derive macro `Queryable` (in Nightly builds, run with -Z macro-backtrace for more info)

error: This derive can only be used on non-unit structs
  --> tests/fail/derive/empty_struct.rs:25:10
   |
25 | #[derive(QueryableByName)]
   |          ^^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `QueryableByName` (in Nightly builds, run with -Z macro-backtrace for more info)

error: This derive can only be used on non-unit structs
  --> tests/fail/derive/empty_struct.rs:28:10
   |
28 | #[derive(Selectable)]
   |          ^^^^^^^^^^
   |
   = note: this error originates in the derive macro `Selectable` (in Nightly builds, run with -Z macro-backtrace for more info)

error: This derive can only be used on non-unit structs
  --> tests/fail/derive/empty_struct.rs:31:10
   |
31 | #[derive(Associations)]
   |          ^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `Associations` (in Nightly builds, run with -Z macro-backtrace for more info)
